### PR TITLE
feat: middleware to verify access token

### PR DESCRIPTION
**Pull request by khoans**
## Description
Middleware to navigate accessToken from LMS client to LMS server for verification. Accessible to any module

### Notion Task
Please include a link to the notion task that this pull request is related to.
- [GEN-388](https://www.notion.so/68233b1464534b7d82e15867c27a7986?v=10575e2b484449898b0482071a4f1a35&p=7b6f04758f3340aa9b6c18a959978eaa&pm=s)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires the database schema to be updated.
- [ ] This change create or update API endpoints.

### General
- [x] I have performed a self-review of my own code.
- [ ] This change has been tested locally to ensure it pass acceptance criterias.
- [x] All conflicts have been resolved.

### Database Changes (if applicable)
- [ ] I have added a migration file to update the database schema.
- [ ] I have updated the database schema ERD in this [link](https://drive.google.com/file/d/1PIRXgwItsAAR-MePVI0IXLRpVyzC3cU6/view?usp=sharing).
- [ ] I have log all change in the log file.
- [ ] All breaking changes have been communicated to the team and reviewed by PO/PM.

### API Changes (if applicable)
- [ ] I have updated the API Spec in this [link](https://docs.google.com/document/d/1WdZVLshSQK6OIrvA4hru1lXAjbDNo5J1AguMZvRaz2g/edit?usp=drive_link).
- [ ] I have update the Postman Collection in this [link](https://www.postman.com/universal-meteor-717123/workspace/manhattan/collection/2573019-7ceeff4f-4d49-405e-8a8c-6dc813d91bad?action=share&creator=2573019).

## Notes
Any other relevant information that would help the reviewer.
Git wiki: https://github.com/FUNiX-Tech/Portal/wiki/Guide:-How-to-use-authentication-middleware-in-portal_student_management

### CODE REVIEW

File: portal-addons/portal_student_management/middlewares/middlewares.py [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+        # 1. Extract access token from cookie
+        access_token = http.request.httprequest.cookies.get("accessToken")
+
+        print("accessToken", access_token)
+        # 2. Check access token exists
+        if not access_token:
+            return json_error("Access token not found", status=400)
+
+        # 3. Construct API request
+        url = "https://test-xseries.funix.edu.vn/api/user/v1/user-info"
```
Comment (DAN3002) [inline portal-addons/portal_student_management/middlewares/middlewares.py:18]: Đoạn này đang fix cứng domain của LMS, anh dùng cái config của anh @hoangbnmfx19875 để có thể setting cái domain này nhé ạ. Sau này mình cũng sẽ tách ra các môi trường nữa.

**Review by DAN3002 — APPROVED** [review]
Cần dùng domain LMS đã được thiết lập ở config

**Review by DAN3002 — CHANGES_REQUESTED** [review]
Cần sửa lại phần domain LMS

File: portal-addons/portal_student_management/middlewares/middlewares.py (hunk 1 above)
Comment (khoans) [reply portal-addons/portal_student_management/middlewares/middlewares.py:18]: ok để mình trao đổi lại với Hoàng,

**Comment by khoans** [comment]
Domain LMS đã được thay bằng service key thay vì code cứng

**Pull request closed without merging**